Decred Wallet/Features/Security/PasswordSetupViewController.swift

```diff
@@ -71,32 +104,58 @@ class PasswordSetupViewController: SecurityBaseViewController, UITextFieldDelega
         return self.validatePasswordsAndProceed()
     }
     
-    @IBAction func onOkTapped(_ sender: Any) {
+    @IBAction func cancelTapped(_sender: UIButton){
+        self.dismiss(animated: true)
+    }
+    
+    @IBAction func createTapped(_ sender: UIButton){
+        createBtn.isEnabled = false
         _ = self.validatePasswordsAndProceed()
     }
     
+    
     func validatePasswordsAndProceed() -> Bool {
         let password = self.tfPassword.text ?? ""
         if password.length == 0 {
             self.showMessageDialog(title: LocalizedStrings.error, message: LocalizedStrings.emptyPasswordNotAllowed)
+            createBtn.isEnabled = true
             return false
         }
         
         if self.tfPassword.text != self.tfConfirmPassword.text {
             self.showMessageDialog(title: LocalizedStrings.error, message: LocalizedStrings.passwordsDoNotMatch)
+            createBtn.isEnabled = true
             return false
         }
         
         // only quit VC if not part of the SecurityVC tabs
         if self.tabBarController == nil {
-            if self.isModal {
-                self.dismiss(animated: true, completion: nil)
-            } else {
-                self.navigationController?.popViewController(animated: true)
-            }
+            self.close()
         }
         
         self.onUserEnteredPassword?(self.tfPassword.text!)
         return true
     }
+    
+    private func close(){
+        if self.isModal {
+            self.dismiss(animated: true, completion: nil)
+        } else {
+            self.navigationController?.popViewController(animated: true)
+        }
+    }
+    
+    private func setupInterface(){
```

Comment:
Whitespace before opening brace. This function should be moved closer to where it is called.

Reply:
resolved in ab83980